### PR TITLE
RegisterMetadata: add missing code for new metadata

### PR DESCRIPTION
**Pull request by xuechen417**
This PR is going to add the missing support of new metadata in the
functions used for glue shader in PalMetadata, such as `mergeFromBlob`
and `getVsEntryRegInfo`. Besides, we add the check to prevent going
through old metadata path if new style is adopted.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2949312ddaa40dd504350d630232ea889312e46c
<details><summary>CTS tests (Failed: 0/213076)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40426/71028 (56.9%)</li>
<li>Failed: 0/71028 (0.0%)</li>
<li>Not Supported: 30602/71028 (43.1%)</li><li>Warnings: 0/71028 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40464/71018 (57.0%)</li>
<li>Failed: 0/71018 (0.0%)</li>
<li>Not Supported: 30554/71018 (43.0%)</li><li>Warnings: 0/71018 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36471/71030 (51.3%)</li>
<li>Failed: 0/71030 (0.0%)</li>
<li>Not Supported: 34559/71030 (48.7%)</li><li>Warnings: 0/71030 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by xuechen417** [comment]
V1: rebase.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 60783d392c843e381f36afbf32653b411b61d285
<details><summary>CTS tests (Failed: 1/213062)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40426/71014 (56.9%)</li>
<li>Failed: 0/71014 (0.0%)</li>
<li>Not Supported: 30588/71014 (43.1%)</li><li>Warnings: 0/71014 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40463/71018 (57.0%)</li>
<li><details><summary>Failed: 1/71018 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.memory.pipeline_barrier.all.65536_vertex_buffer_stride_4
Stack trace: Timeout
```
</p></details></li>

<li>Not Supported: 30554/71018 (43.0%)</li><li>Warnings: 0/71018 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36471/71030 (51.3%)</li>
<li>Failed: 0/71030 (0.0%)</li>
<li>Not Supported: 34559/71030 (48.7%)</li><li>Warnings: 0/71030 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 60783d392c843e381f36afbf32653b411b61d285
<details><summary>CTS tests (Failed: 0/213076)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40426/71028 (56.9%)</li>
<li>Failed: 0/71028 (0.0%)</li>
<li>Not Supported: 30602/71028 (43.1%)</li><li>Warnings: 0/71028 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40464/71018 (57.0%)</li>
<li>Failed: 0/71018 (0.0%)</li>
<li>Not Supported: 30554/71018 (43.0%)</li><li>Warnings: 0/71018 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36471/71030 (51.3%)</li>
<li>Failed: 0/71030 (0.0%)</li>
<li>Not Supported: 34559/71030 (48.7%)</li><li>Warnings: 0/71030 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/state/PalMetadata.cpp [hunk 1]
```diff
@@ -1108,26 +1243,47 @@ unsigned PalMetadata::getFragmentShaderBuiltInLoc(unsigned builtin) {
...
-      {mmSPI_SHADER_PGM_RSRC1_ES, CallingConv::AMDGPU_ES}, {mmSPI_SHADER_PGM_RSRC1_GS, CallingConv::AMDGPU_GS},
-      {mmSPI_SHADER_PGM_RSRC1_VS, CallingConv::AMDGPU_VS}, {mmCOMPUTE_PGM_RSRC1, CallingConv::AMDGPU_CS}};
-
-  for (unsigned i = 0; i < hwShaderStageCount; ++i) {
-    auto entry = m_registers.find(m_document->getNode(shaderTable[i].first));
-    if (entry != m_registers.end())
-      return shaderTable[i].second;
+//
+// @param hwStageId :
+unsigned PalMetadata::getCallingConventionForFirstHardwareShaderStage(std::string &hwStageName) {
```
Comment (nhaehnle) [inline lgc/state/PalMetadata.cpp:1248]: The @param comment needs to be updated, and state explicitly that it's only filled in with the new register field format.

File: lgc/state/PalMetadata.cpp [hunk 2]
```diff
@@ -1108,26 +1243,47 @@ unsigned PalMetadata::getFragmentShaderBuiltInLoc(unsigned builtin) {
...
+        {mmSPI_SHADER_PGM_RSRC1_LS, CallingConv::AMDGPU_LS}, {mmSPI_SHADER_PGM_RSRC1_HS, CallingConv::AMDGPU_HS},
+        {mmSPI_SHADER_PGM_RSRC1_ES, CallingConv::AMDGPU_ES}, {mmSPI_SHADER_PGM_RSRC1_GS, CallingConv::AMDGPU_GS},
+        {mmSPI_SHADER_PGM_RSRC1_VS, CallingConv::AMDGPU_VS}, {mmCOMPUTE_PGM_RSRC1, CallingConv::AMDGPU_CS}};
+
+    for (unsigned i = 0; i < hwShaderStageCount; ++i) {
+      auto entry = m_registers.find(m_document->getNode(shaderTable[i].first));
+      if (entry != m_registers.end())
+        return shaderTable[i].second;
+    }
+    return CallingConv::AMDGPU_CS;
```
Comment (nhaehnle) [inline lgc/state/PalMetadata.cpp:1277]: The diff could be simplified by not using if/else. The true case always returns from the function, so there's no need to put the original code into an else block.

File: lgc/state/PalMetadata.cpp (hunk 1 above)
Comment (xuechen417) [reply lgc/state/PalMetadata.cpp:1248]: Done.

File: lgc/state/PalMetadata.cpp (hunk 2 above)
Comment (xuechen417) [reply lgc/state/PalMetadata.cpp:1277]: Got it. Fixed. Make the same modification on `isWave32`.

**Comment by amdvlk-admin** [comment]
### Test summary for commit b1dd7b735cbbd0285696d5b6df5591bdd66e70d0
<details><summary>CTS tests (Failed: 0/213062)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40426/71014 (56.9%)</li>
<li>Failed: 0/71014 (0.0%)</li>
<li>Not Supported: 30588/71014 (43.1%)</li><li>Warnings: 0/71014 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40464/71018 (57.0%)</li>
<li>Failed: 0/71018 (0.0%)</li>
<li>Not Supported: 30554/71018 (43.0%)</li><li>Warnings: 0/71018 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36471/71030 (51.3%)</li>
<li>Failed: 0/71030 (0.0%)</li>
<li>Not Supported: 34559/71030 (48.7%)</li><li>Warnings: 0/71030 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by xuechen417** [comment]
V2: remove unused class member m_pipelineShaders that causes clang warning.

**Comment by amdvlk-admin** [comment]
### Test summary for commit be58c51d3b250ca77d6cea5a6f983c19d9dcdd99
<details><summary>CTS tests (Failed: 0/213062)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40426/71014 (56.9%)</li>
<li>Failed: 0/71014 (0.0%)</li>
<li>Not Supported: 30588/71014 (43.1%)</li><li>Warnings: 0/71014 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40464/71018 (57.0%)</li>
<li>Failed: 0/71018 (0.0%)</li>
<li>Not Supported: 30554/71018 (43.0%)</li><li>Warnings: 0/71018 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36471/71030 (51.3%)</li>
<li>Failed: 0/71030 (0.0%)</li>
<li>Not Supported: 34559/71030 (48.7%)</li><li>Warnings: 0/71030 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
Thanks!

**Pull request merged**